malariagen_data/anoph/fst.py

```diff
@@ -493,7 +493,9 @@ def plot_pairwise_average_fst(
         **kwargs,
     ):
         # setup df
-        cohort_list = np.unique(fst_df[["cohort1", "cohort2"]].values)
+        cohort_list = pd.unique(
+            [c for cl in fst_df[["cohort1", "cohort2"]].values for c in cl]
```

Comment:
The for loops in this line look a little funky. What is it supposed to be doing?

Reply:
The goal is to get the list of all cohorts in the same order they were defined in the cohort dictionary. `plot_pairwise_average_fst` does not have the list of cohorts or the dictionary as an input so we have to access it in another way. `fst_df[["cohort1", "cohort2"]]` is a list of list each one containing two cohort names. The last cohort of the dictionary is missing from the `cohort1` column and the first is missing from `cohort2`, hence why they are both accessed. `np.unique` is able to take a list of list and return every unique value but it sorts them alphabetically breaking the dictionary's order. `pd.unique` keeps the order but can't deal with a list of lists. The for loops are thus used to go from a list of list of cohorts to a list of cohorts.

Reply:
Thanks @jonbrenas. What about:

```
pd.unique(fst_df[["cohort1", "cohort2"]].values.flatten())
```

Reply:
Nested list comprehensions always bake my noodle!

I _think_ it's equivalent to this:

```python
cohort_pairs = fst_df[["cohort1", "cohort2"]].values
flattened_cohorts = []
for cohort_pair in cohort_pairs:
	for cohort in cohort_pair:
		flattened_cohorts.append(cohort)
```

Or like Alistair suggested:
```python
list(fst_df[["cohort1", "cohort2"]].values.flatten())
```